app/javascript/components/applications/application_form/PaperApplicationForm.js

```diff
@@ -53,6 +54,22 @@ class PaperApplicationForm extends React.Component {
       errors.alternate_contact.last_name = validate.isPresent('Please enter a Last Name')(values.alternate_contact.last_name)
       errors.alternate_contact.email = validate.isValidEmail('Please enter a valid Email')(values.alternate_contact.email)
     }
+
+    // Secondary preference application member validation: Check if selected app member value is still a valid natural key.
+    if (values.preferences) {
+      // Combine household members with primary applicant to get list of all the natural keys
+      const allMembers = [values.applicant].concat(values.household_members ? values.household_members : [])
```

Comment:
I think we need to filter `allMembers` and use them then they are valid. Right now form breaks when I click `Add Preference` when HH member is missing date or when I delete it.

Reply:
excellent bug hunting! will investigate